### ISSUE TITLE
Estimate gas return value is wrong on crab

### ISSUE THREAD

**Issue by hujw77** [darwinia-network/darwinia-common#1145]
## script
```
~ curl -fsS https://crab-rpc.darwinia.network -d '{"id":1,"jsonrpc":"2.0","method":"eth_estimateGas","params":[{"data":"0x341b989c0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000a2a0300010300010100000000000000030000000000000000000000000000061f2a030001030001010000000000000003000000000000000000000000000000942a0300010300010100000000000000030000000000000000000000000000008d2a030001030001010000000000000003000000000000000000000000000000852a030001030001010000000000000003000000000000000000000000000000842a0300010300010100000000000000030000000000000000000000000000007b2a030001030001010000000000000003000000000000000000000000000000632a030001030001010000000000000003000000000000000000000000000000582a030001030001010000000000000003000000000000000000000000000000542a03000103000101000000000000000300000000000000000000000000000052","from":"0xe32c5673b152571E57976bF3A75068917DA2Da9C","to":"0x4E319297F77EA8099520F3467a083050090a1d81"}]}' -H 'Content-Type: application/json'
4115012357
```

**Comment by AurevoirXavier** [comment]
Could you reproduce on Pangolin?
Otherwise, this issue should be moved to darwinia.
They are using different DVM versions.

**Comment by hujw77** [comment]
Because of EVO is deployed on crab, so I could not reproduce the bug on Pangolin.

**Comment by AsceticBear** [comment]
It works after the client upgrade to the v0.12.0 version. Reopen it if happened again.

**Comment by hujw77** [comment]
## claim evo land resource, 20 lands
```shell
+ curl -fsS https://http-mainnet-node.huobichain.com -d '{"id":1,"jsonrpc":"2.0","method":"eth_estimateGas","params":[{"data":"0x341b989c000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000162a040001040001010000000000000004000000000000000000000000000001292a040001040001010000000000000004000000000000000000000000000001a02a040001040001010000000000000004000000000000000000000000000001b92a040001040001010000000000000004000000000000000000000000000001cb2a0400010400010100000000000000040000000000000000000000000000021b2a040001040001010000000000000004000000000000000000000000000002222a040001040001010000000000000004000000000000000000000000000002262a0400010400010100000000000000040000000000000000000000000000022e2a0400010400010100000000000000040000000000000000000000000000024e2a040001040001010000000000000004000000000000000000000000000002522a040001040001010000000000000004000000000000000000000000000002592a040001040001010000000000000004000000000000000000000000000002612a040001040001010000000000000004000000000000000000000000000002a32a040001040001010000000000000004000000000000000000000000000003bd2a040001040001010000000000000004000000000000000000000000000003da2a040001040001010000000000000004000000000000000000000000000003e22a040001040001010000000000000004000000000000000000000000000003e92a040001040001010000000000000004000000000000000000000000000003fb2a040001040001010000000000000004000000000000000000000000000004192a040001040001010000000000000004000000000000000000000000000004232a0400010400010100000000000000040000000000000000000000000000042b2a0400010400010100000000000000040000000000000000000000000000042f","from":"0xE187F4fD434C4e1b4c2fDA832DF3c0B70c3E9a03","to":"0x34Ab84986f0a676A622a15eEdFBe8d344AE66A89"}]}' -H 'Content-Type: application/json'
4508387
```

**Comment by hujw77** [comment]
## claims evo land resource on crab chain, 11 lands
```shell
+ curl -fsS https://crab-rpc.darwinia.network -d '{"id":1,"jsonrpc":"2.0","method":"eth_estimateGas","params":[{"data":"0x341b989c0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000b2a0300010300010100000000000000030000000000000000000000000000002e2a030001030001010000000000000003000000000000000000000000000000302a030001030001010000000000000003000000000000000000000000000000312a030001030001010000000000000003000000000000000000000000000000352a030001030001010000000000000003000000000000000000000000000000482a0300010300010100000000000000030000000000000000000000000000005a2a030001030001010000000000000003000000000000000000000000000000672a0300010300010100000000000000030000000000000000000000000000006f2a030001030001010000000000000003000000000000000000000000000000722a030001030001010000000000000003000000000000000000000000000000832a03000103000101000000000000000300000000000000000000000000000087","from":"0x7EEF076Bc5C57afd42Ac3CBF3E29B13A7BAF4CaC","to":"0x4E319297F77EA8099520F3467a083050090a1d81"}]}' -H 'Content-Type: application/json'
41962810
```

**Comment by hujw77** [comment]
## claims evo land resource on crab chain, 5 lands
```shell
+ curl -fsS https://crab-rpc.darwinia.network -d '{"id":1,"jsonrpc":"2.0","method":"eth_estimateGas","params":[{"data":"0x341b989c000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000052a0300010300010100000000000000030000000000000000000000000000002e2a030001030001010000000000000003000000000000000000000000000000302a030001030001010000000000000003000000000000000000000000000000312a030001030001010000000000000003000000000000000000000000000000352a03000103000101000000000000000300000000000000000000000000000048","from":"0x7EEF076Bc5C57afd42Ac3CBF3E29B13A7BAF4CaC","to":"0x4E319297F77EA8099520F3467a083050090a1d81"}]}' -H 'Content-Type: application/json'
5231085
```

**Comment by AsceticBear** [comment]
I just test those cases in the newly crab v0.12.1 version:
1. For claims evo land resource on crab chain, 11 lands, the result is 2511354
2. For claims evo land resource on crab chain, 5 lands, the result is 1200641

After compared with the result from heco. Is this result acceptable? @hujw77

**Comment by hujw77** [comment]
Yes, It's reasonable

**Comment by AsceticBear** [comment]
Fixed in crab node v0.12.1
